python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -38,19 +39,21 @@
 
 
 class ResourceDemandScheduler:
-    def __init__(self, provider: NodeProvider,
-                 node_types: Dict[NodeType, NodeTypeConfigDict],
-                 max_workers: int):
+    def __init__(self, provider: NodeProvider, cluster_config: Dict[str, Any]):
         self.provider = provider
-        self.node_types = node_types
-        self.max_workers = max_workers
+        self.cluster_config = copy.deepcopy(cluster_config)
+        self.node_types = self.cluster_config.get("available_node_types")
+        if "available_node_types" in self.cluster_config:
+            self.is_legacy_yaml = False
+        else:
+            self.is_legacy_yaml = True
 
     def get_nodes_to_launch(
             self, nodes: List[NodeID], pending_nodes: Dict[NodeType, int],
             resource_demands: List[ResourceDict],
             usage_by_ip: Dict[NodeIP, ResourceDict],
-            pending_placement_groups: List[PlacementGroupTableData]
-    ) -> Dict[NodeType, int]:
+            pending_placement_groups: List[PlacementGroupTableData],
+            static_node_resources: List[ResourceDict]) -> Dict[NodeType, int]:
```

Comment:
Is this excluding the head node?

Reply:
I added a TODO to fix it to look only at the head node.